projects/RabbitMQ.Client.OpenTelemetry/TraceProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using OpenTelemetry.Context.Propagation;
+using RabbitMQ.Client;
+using RabbitMQ.Client.OpenTelemetry;
+
+namespace OpenTelemetry.Trace
+{
+    public static class OpenTelemetryExtensions
+    {
+        internal static TextMapPropagator s_propagator = Propagators.DefaultTextMapPropagator;
+
+        public static TracerProviderBuilder AddRabbitMQ(this TracerProviderBuilder builder,
+            RabbitMQOpenTelemetryConfiguration configuration)
+        {
+            if (configuration.PropagateBaggage)
```

Comment:
any reason not to allow user to provide a preconfigured propagator instance?
E.g. users could want to use [B3 propagation](https://github.com/openzipkin/b3-propagation) or [XRay AWS propagator](https://github.com/aws-observability/aws-otel-dotnet)

Reply:
No reason bu my lack of knowledge :). Are you then referring to these as alternatives to the tracestate propagator? Are there examples for me to look at?

Reply:
I mean you're forcing a specific propagator - a `CompositeTextMapPropagator` or `TraceContextPropagator` which is not necessary.

I believe you should just use the `DefaultTextMapPropagator`. 
Users can configure it however they want to, check out  https://github.com/open-telemetry/opentelemetry-dotnet/blob/9b246750ab76e94986fc433d5c7af046a137e477/src/OpenTelemetry.Extensions.Propagators/README.md?plain=1#L29

If I understand the intent correctly you want to users to be able to disable baggage propagation - any specific reason?

Reply:
> I mean you're forcing a specific propagator - a `CompositeTextMapPropagator` or `TraceContextPropagator` which is not necessary.
> 
> 
> 
> I believe you should just use the `DefaultTextMapPropagator`. 
> 
> Users can configure it however they want to, check out  https://github.com/open-telemetry/opentelemetry-dotnet/blob/9b246750ab76e94986fc433d5c7af046a137e477/src/OpenTelemetry.Extensions.Propagators/README.md?plain=1#L29

Ahh gotcha! Saw that the DefaultTextMapPropagator was a Noop so I thought this was the way. I'll change it :)
> 
> 
> If I understand the intent correctly you want to users to be able to disable baggage propagation - any specific reason?

Nope. I'll remove it. Does it make sense to then have a composite propagator with the default textmap propagator and the baggage propagator or do users normally configure the baggage propagator themselves?

Reply:
The otel sdk should set the default one to the [composite](https://github.com/open-telemetry/opentelemetry-specification/blob/a03382ada8afa9415266a84dafac0510ec8c160f/specification/context/api-propagators.md?plain=1#L310).

Not completely sure what sets `OpenTelemetry.Context.Propagation.Propagators.DefaultTextMapPropagator`, but I see it's set after OTel SDK is configured:)